client/src/pages/ProfileDetails/BookingForm/BookingForm.tsx

```diff
@@ -1,69 +1,108 @@
-import useStyles from './useStyles';
-import { Box, Typography, Select, MenuItem, OutlinedInput, Grid, Button, Card } from '@material-ui/core';
+import { useEffect, useState, useContext } from 'react';
+import { useParams } from 'react-router-dom';
+import { Box, Typography, Select, MenuItem, Grid, Button, Card } from '@material-ui/core';
 import Rating from '@material-ui/lab/Rating';
 import { useFormik } from 'formik';
-
-const times = [...Array(24).keys()].map((number) => {
-  if (number === 0) {
-    return {
-      time: '12:00 AM',
-      value: number,
-    };
-  } else if (number < 12) {
-    return {
-      time: number.toString() + ':00 AM',
-      value: number,
-    };
-  } else if (number === 12) {
-    return {
-      time: number.toString() + ':00 PM',
-      value: number,
-    };
-  } else {
-    return {
-      time: (number - 12).toString() + ':00 PM',
-      value: number,
-    };
-  }
-});
+import { SnackBarContext } from '../../../context/useSnackbarContext';
+import { AuthContext } from '../../../context/useAuthContext';
+import fetchProfile from '../../../helpers/APICalls/fetchProfile';
+import createRequestNotification from '../../../helpers/APICalls/createRequestNotification';
+import useStyles from './useStyles';
+import { times } from './times';
 
 const BookingForm = (): JSX.Element => {
+  const { loggedInUser } = useContext(AuthContext);
+  const { updateSnackBarMessage } = useContext(SnackBarContext);
+  const { id } = useParams<{ id?: string }>();
   const classes = useStyles();
+  const [availableTime, setAvailableTime] = useState([
+    {
+      date: 'Monday-default',
+      startTime: '8:00 AM - default',
+      endTime: '10:00 PM - default',
+      available: true,
+    },
+  ]);
+  const [availableStartHours, setAvailableStartHours] = useState([
+    {
+      time: '12:00 AM - default',
+      value: 0,
+    },
+  ]);
+  const [availableEndHours, setAvailableEndHours] = useState([
+    {
+      time: '11:00 PM - default',
+      value: 23,
+    },
+  ]);
+
+  useEffect(() => {
+    if (id) {
+      fetchProfile(id).then((data) => setAvailableTime(data.success.profile.availableTime));
+    }
+  }, [id]);
 
   const formik = useFormik({
     initialValues: {
-      startDate: '',
+      startDate: 'placeholder',
```

Comment:
Why don't we use the values from the default state above?

Ditto below!

Reply:
I have the values separated since I don't want to pre-fill the form with any values returned by the fetchProfile call. Each of the initialValues maps to a MenuItem that displays the relevant placeholder for each input on the form.